app/javascript/components/remove-generic-item-modal.jsx

```diff
@@ -29,19 +30,30 @@ export const removeItems = (items, apiUrl, asyncDelete, redirectUrl) => {
         miqSparkleOff();
       } else {
         apiData.forEach(item => {
+          let flash = {};
           if (item.result === 'success') {
-            miqFlashLater({message: sprintf(deleteMessage, item.name)});
+            flash = {message: sprintf(deleteMessage, item.name), level: 'success'};
           } else if (item.result === 'error' && items.length > 1) {
-            miqFlashLater({message: sprintf(__('Error deleting item "%s": %s'), item.name, parseApiError(item.data)), level: 'error'});
+            flash = {message: sprintf(__('Error deleting item "%s": %s'), item.name, parseApiError(item.data)), level: 'error'};
+          }
+          if (typeof(treeSelect) == undefined) {
+            miqFlashLater(flash);
+          } else {
+            flashArray.push(flash);
           }
         });
       }
       return apiData;
     })
     .then((apiData) => {
       if (items.length > 1 || (items.length === 1 && apiData[0].result === 'success')) {
-        window.location.href = redirectUrl; // '/catalog/explorer?report_deleted=true';
-        miqSparkleOff();
+        if (typeof(treeSelect) == undefined) {
+          window.location.href = redirectUrl;
+          miqSparkleOff();
+        } else {
+          sendDataWithRx({ controller: 'reportDataController', type: 'gtlUnselectAll' });
+          miqAjax(`tree_select?id=${treeSelect}`).then(() => flashArray.forEach((flash) => add_flash(flash.message, flash.level)));
```

Comment:
You just use `miqFlashLater` in both cases and do `miqFlashSaved()` here

Reply:
Yup, fixed, thanks.